roachtest: sqlsmith/setup=empty/setting=no-mutations failed

[(roachtest).sqlsmith/setup=empty/setting=no-mutations failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2283744&tab=buildLog) on [release-20.2@4ecea82f7dfaf5a4edd4d7c7ffad4a4e7c6c4cce](https://github.com/cockroachdb/cockroach/commits/4ecea82f7dfaf5a4edd4d7c7ffad4a4e7c6c4cce):

```
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=empty/setting=no-mutations/run_1
	sqlsmith.go:172,sqlsmith.go:202,test_runner.go:754: error: pq: internal error: runtime error: index out of range [0] with length 0
		stmt:
		WITH
			with_6634 (col_36481, col_36482)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								('-71 years -5 mons -28 days -13:18:47.333491':::INTERVAL, 2268815589:::OID),
								(
									(experimental_follower_read_timestamp()::TIMESTAMPTZ::TIMESTAMPTZ - '1970-01-12 13:23:56.000459':::TIMESTAMP::TIMESTAMP)::INTERVAL,
									677293670:::OID
								)
						)
							AS tab_21274 (col_36481, col_36482)
				),
			with_6635 (col_36483)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									'010200000002000000AE69E2DF4965EEC79470F0EC59D1C2C70A3B77CAE04FE1C7065C8F6BEF23EAC7':::GEOMETRY
								),
								(
									st_scale('01060000000A00000001030000000100000006000000446F63920DC7C4C7D1E9AD2983B0E7C760D9FE2EE82DED47EE00953915E6E2C7CC506D178328DBC776E94D8252FCE947CE9D8C529C44E2C7A81CB31E2DBAE147028B535B669BD0C7002A4074FDC4B347446F63920DC7C4C7D1E9AD2983B0E7C7010300000001000000080000003CE36CEBADF8D1C740D2D8D4309AC9475A4E30B5B8E8EAC740C4EE3DA42BCCC7E044019B2D98CBC700511AF5DD699F47C4DA1CAE8240E347546576FB3F22EAC7BE320E2E2B54EA473EA024865E04ED4780E548D5EF999F474ACD40368CE3EC47E037E56ECD2FCCC71C0AAD8DB5EBE4473CE36CEBADF8D1C740D2D8D4309AC94701030000000100000006000000DC61715654CDEBC750DF90FC903ADDC72000AE95013CDB471C39F1EDB050E6C730DE54EF32FCED470E1FC01DFE10E3C7A9F4E558E016E3C7A8743441FCCCE4470E0D75E84E9BDDC7C0B737E51CD2CC47DC61715654CDEBC750DF90FC903ADDC70103000000010000000600000058ED539A0B52C8C7285AF1868EDEC847AC7735771D7FD5C7F401BB2A974AC0C7660C918E31AEED4760AC98417CBAC6C7008C467AE35FD9471C8EA29CA889EF4770E152489B07D8C7E4A118F4455EE34758ED539A0B52C8C7285AF1868EDEC84701030000000100000006000000CE45CC67F333E6C7209339F1F9E2EAC7B85F51494E1ED0C7464684B796EDE6C70041AA71C150BC4732B7576ABBF9E6C7800FBBE232ACE647DCBA28148C24C2C76CE6A3DD604DEA473CC1EDF280B6DD47CE45CC67F333E6C7209339F1F9E2EAC70103000000010000000D000000CAAF1F67C91BECC740C9F69CBB19D9C7DC264969586DE5C71AC692CDB3E0D8C7A00443099295CAC7D8F4972FF067D8C79034C5B26D2ACEC7E8D5C64F53DEE4C7100CDE6E748DADC7F5114F0E0AC9E5C7ECE147548122E647EC0218C7591DE8C732E1E99F8236E547844C448F4462C9C700F16C18FCC3D947C0F7C2EAA393BE47946817BE1FB9D24720767AF2B272E047E8947AF61BF9C14716CDC5A72970EB475E380FEB7249D4C760B305C720C6BC478E9DD6CBBE2DE2C75026258178B8B747CAAF1F67C91BECC740C9F69CBB19D9C70103000000010000000600000098D4A1E40547E3C7C03A4887AC77D6C720CEFDCEF604A3C7A619F0A3AA63E8C7449407F50AC2D54704DDF48D3897CEC7D4DB97E65CABED47F0D7519750E5CCC7E8BA6628CB4EB6C7CC91B0947983E04798D4A1E40547E3C7C03A4887AC77D6C70103000000010000000B00000021FA3395A442E9C7D11BCF5C063BE9C76C3D3E2A8DBFDFC7A63A813E92E1E5C70450902E969EE8471CB809EF3D2EE3C750C5F763BA76E54752FB21BC51F4E0C7AC5107C32E64ED477A6F50011C33DAC798D4C2458A71D6474C58C51766F9C4C7080EB0449FF0CC47C0664A00BDFEB04738219908DB15BFC7D0998D5505ADC0478A61B69F4838E9C71489D9BED19FD547AC9A9747821CE8C7C093ED8634879EC721FA3395A442E9C7D11BCF5C063BE9C70103000000010000000B000000B12D69DA856EEFC732B3FB8FA375D0C7368A13E9758AECC7A8E98E5890FDEEC708CB5B9BE1D7B4C758799A03EEF1C7C7E85834045AF9E1471BDC9BE26AD7EBC790218294FFB1D2477ECED05E1825D6C73833CBD8515FE54758122A8190C2BDC748D05C7A730EC447C04871152CE3B847401B6F0D49A09D47DCAAC1141494DD479C8848796C97C2C7481EBB39DA26E5479889C976EBE9E7C7205F6113B572CF47B12D69DA856EEFC732B3FB8FA375D0C701030000000100000008000000DE6DC3E4CA3DE8C770A55A1636BFB6C78687A89A92DEE8474FEFAB6E80E2E9C798FEF0D3E594DC4764C20E7FFE8DD8C7DC9288BD3B9FD0472CEB5C45EC5DDD47FA6FC0EA5C33D8C7AEA6408CDE91E1475A96677A8C44E2C718A1CA65C775E94782A6D244CDABD3C7306050B66326B747DE6DC3E4CA3DE8C770A55A1636BFB6C7':::GEOMETRY::GEOMETRY, '0101000000000000000000F87F000000000000F87F':::GEOMETRY::GEOMETRY)::GEOMETRY
								),
								(
									'0107000000080000000106000000070000000103000000010000000A000000AA0919665728E2C772D0A4C1D4B8E2C7C8A54E00727CCE476C240F44E343DBC7108935677E8FD24770E1A2C4F5EDDCC7E620D2E04A4EE647D62FE62A00ABE6C778F5279DCA04E447409F13AD6BDED2C730E182948044E647C04F49AD8125BD47600EA296EEB1DB47748C588A727CD1475E8D723C1184E1474239E5C535DCEF4779A570C57376EBC71E66918496C5EC47AA0919665728E2C772D0A4C1D4B8E2C70103000000010000000800000090B334A09CD5BEC73073C7E27059CDC7FE50A99FED5CD1C73C5634A6573EE9C7600E3C00C65AA947F0BE231345ADE8C7DE0E86148EA2EA4714F96E79EA6FC1C75A4683ED592AE647F85DACB1B2C5C947468725140BC2EF470A26F3651830E647801144E1ACC0E947A01F675407D6EA4790B334A09CD5BEC73073C7E27059CDC7010300000001000000040000007415F9F2C145D54718A4B5829FD2DAC7389E8AD256F0DF474E575BEDE3E2D8C7A8A34E36A4D5D6472035428180A5AEC77415F9F2C145D54718A4B5829FD2DAC701030000000100000004000000E618DDE1858FD0C7F34C0018C921E7C720438881DA16CDC71CFEDA3B7F66C4C705F37C7342C5E6C78097161F2BCCA847E618DDE1858FD0C7F34C0018C921E7C7010300000001000000060000003484D509ECC9E4C747EB3BB990B4EDC7B0DE70D64710C64714C031BB8CC9E5C7B4E4307BC9C9EF47E8D7A48083AAE247E02637E2529EB34744E858CBB171E64720663314263FBE4700F0549EE4E34A473484D509ECC9E4C747EB3BB990B4EDC7010300000001000000070000003899A5561F5DDAC77826EE6B0547C44734C935C618CFE5C7B0387A4E5E72A3C772EC221505B2E5C7873DDA0692F0E5C7E018B408C395ACC7303D7E78F4A6BA47A8E454925510DC4738A7681DBB26E5476C885AC8C87DCDC78A2A24F386F8EF473899A5561F5DDAC77826EE6B0547C4470103000000010000000A0000009663C06DBB71E7C7D46A55227064CAC74132794FD91AE1C7F095383B2E5EE9C7E0A1523FB75CB24760829AA67A43C6C7B4BD19597593E34794FF4CC44B1BD74790B20A8F6476DC4786F7DA579212EA477009C579C3E3CE47BC229B0EE86BEA4782961F58382CD9C74CF08A5AC22FE54744F5C9EBE999CDC724892999BE38D547CA245DF60CDCE9C752C68954CFBDE1479663C06DBB71E7C7D46A55227064CAC701030000000100000004000000A8CAB46D3F7ED1474D7CD91A6C22EEC756FDF88A3B47E547448FB603A3C3D8472A8A9ADD9459E3C7F0D68E364712DE47A8CAB46D3F7ED1474D7CD91A6C22EEC7010200000002000000A0EFEE2CA23BE547FC8E7944DA59E6C7F0DEDA614576D6470006A6A358BF874701020000000300000020FDB98B8CDFA3C7076C904B4499E6C728C8EAD2ADD8E3474896BD6BF306DCC72C0ABEDCE67DD447CC3C936AA449DF47010200000004000000548FA789B6FCDB47E8397C906BEDE3C79A7B97302FAAE047DC5B02362CABD447E01EF0C013B4E447F8F51CE237D3EB472020844AB0A4A6470A618958CB77E24701040000000400000001010000000EE78C8E8983EEC72233127E7ABDE74701010000008600643CC22CE04718CEB8BD2B35CF47010100000060A11644F833CB4782BE4A8DC84DE9C70101000000A62205A46C1EE3C75056FAA81433C44701060000000A0000000103000000010000000900000054279009BB11E5C7FA320F6C06E5E1C78DF1F244C308E5C742B186DE2EE6ECC7D619BA790B6FD8C74AD6729D3394E7C700A4F7780088BDC7FEA425AFFB05ECC70A8A87E891D7E9471C1BA9EAD7DCC3C746E45D0A622CE347460D6E02BE92E147D89D23F71578B1C756EDA788C14AE24760252DE23074C0C73CFB275353A1E24754279009BB11E5C7FA320F6C06E5E1C70103000000010000000B000000E077E94B3596DCC7DA0A5C0CCC87D5C7E39770B65C66EFC728A0E927F2A9E6C79C15AFDA0393C7C7C53C51FB0AE2EEC7E02E38E91E1A93C7F8843FB5CFD2E2C7BCF1D0148A63D3477384DEDEE0DAEBC7782446B7062ADA47144AFA620CF6E1C7547A038290D0DD47C86F284A1C06B7C780B56E4EA5DECC47880FD99B988ECB4792C951ED2412DAC7683A100F49C7D247D91E3EA73C5BEFC74CE3F90CE7F5D747E077E94B3596DCC7DA0A5C0CCC87D5C70103000000010000000900000060343F6A74C2CAC7AC2C0FEF647FC1C780FAAE3A0A58DBC7BDD770157803E8C7F08BB53E5A3EB8C700CC0D709845D0C700C38E0DFA798F47BCE5519059A8E3C7C242FFAC56CBEF47BC0E4855FF64D047F817FD40ECB5C147F03C067C18B7C2477075B3EF830FDBC70484864452E3D747DA67F436338AE6C788187849C24FCB4760343F6A74C2CAC7AC2C0FEF647FC1C701030000000100000004000000E09DED78F786E64790C869162B25C7474C6FB93CC9EDDB47983146C06D4ED947DE56A6F0EDC6DEC720FDEA74DE47E847E09DED78F786E64790C869162B25C747010300000001000000040000003BBA5A5CE915E9C749DCD641CDDDECC7E454042928B7EC4752B7F2CEC057E4C702887BA42DEBD5C7E05267FA2FDAE3C73BBA5A5CE915E9C749DCD641CDDDECC70103000000010000000D0000007E3CA54F9154EEC790F08B61E5F4BD478D043F1E310CECC796A61746CE79E3C7966F7CC4FEC1EA47C0429905C8E7B147D0D497D4CAEDEE47D8FB2F89E171C4474240107767EFE54750E38456A0E8CE4720FFE1CF1088E9472AF4D5042D20E4471441B084D578E247CC52E8637F17E247BAEDBBD38EF4EA4780C0C63DA815E9477819ADBBAEE2C9478C56E511E874EB47FA50C658565AD1C7A4F7D9265950EC4700AB9EA10D35C3C72642D88D8503E247083D5763F398E6C77C5C07D74EF8DC477E3CA54F9154EEC790F08B61E5F4BD470103000000010000000D000000D3A07B04B4E4EAC7F014D786FA15D5C77C887F6E7F70EEC716DACF53C1C0DEC7FA40DFF64401D8C7682A6534C058CEC7602D7E901808B4476DE54BAB0AF8E6C78C0127E958E1ED47806C7385624B8CC7B0E9334AE0A5CF47743A4282E3C9E04758A2317D5CE6D5473E1E7854A940E347306CD8AE8A8EC6478C85E46D68F9E447C0A522270E9DAC47DA5A8F457830EB475C7E25AD5413D9C79AFD9D9E0A3BEC477DF10F78F292E3C79E362D69FF04E7479E3BE949E5F6E5C7E04E588872D3D647D3A07B04B4E4EAC7F014D786FA15D5C70103000000010000000C000000909FEDFBA469DDC780102DE28746944714343AE19426E0C700DA3BC54EC691C7CA214FAB55E3D8C74000EA606BD287C710BC3A4E7BF4D7C7AEF9B8B9EF85DDC7B49CBE9FCA90E94720D2087F8B6AA0C7AA30858A13B3E947D84F0550BA47D447C0C898938F06A947E0F7C12A321CD247AA65B209A1CCD9C7AADD6C1A2F03E447A240CB065442E7C72A4F43C172DFE14765ECBC702168EAC7D01B2EF8D97CD847467AD2F97FE3E9C7A03DECE6006BD547909FEDFBA469DDC780102DE2874694470103000000010000000B000000AC925790BC04DEC791A430224002EBC760BEEFD49488AE47BC1F126F00FFD5C7A00062F514CBAA472EB8A6813B3ED3C720DDA329B033A847A2DBFA28280AD2C75CA54B2AE662DE4708F94245C822D6C73291983E5815EE478C440BADFB4CDB4720F36E01B873B6C79417FA14ACBEDD47AC2DEF85B194E9C7F092F92419BFC947056C19238120E9C73073F22E992FA2C70EA99B9C044AE1C738F1EAA5656CB0C7AC925790BC04DEC791A430224002EBC701030000000100000006000000BCFF99143985DEC73CE483DA52CFC3C7602727919F53A84795DAD2B782A4ECC7C0FAC8F8CA86D347ECFE6BC16506E647BE4F0CB2B0A6E9C7F867525BE8C6CB473E07B83CF301EDC7C07E08509D7FA647BCFF99143985DEC73CE483DA52CFC3C70103000000010000000D000000E859313EA44EE1C74C57625D95A2D6C7F02D9F4E8DEAA1C73CBC7029F699E0C7502F461045EABE47CA8721C04F88E6C74E612997BBECE3472B601C027AADEBC71A768248783FE847903E27138AAFA0C7324AE349E41DEF47C0F58D3673FE94475A9555435B4AED4728DDC36284C4DB47066C0D7BB9BFE647B61096F1144DEA4768AFD7925E89E54748F0AFB703FAEB4740B0E962A430AFC78417AEE3D0F3EF4761256ADCEB77E1C750939E2F7A2CD847DEE3C48B53BBE9C774C1C9AE560ADA47E859313EA44EE1C74C57625D95A2D6C7':::GEOMETRY
								),
								(
									COALESCE(
										NULL,
										'01060000000700000001030000000100000004000000003556E71FAF88C79ED319C81E61D7C77019DD894F58BC47D880BCE9DE82CBC73008434E0CD0E9C710411344C4D6CBC7003556E71FAF88C79ED319C81E61D7C70103000000010000000B000000E837C18F8C3FC9C7ACC39DA06932DEC76073F39742CAA7C7A4D4A5C4181DE6C77C3D2EDBB0E6D347EC1D239D3C3AD0C75847F7CC11BDE9475C8EE9AEAF66E7C75AE2CD489A10EC47DE195BFFCB61E7C728DD900F14D5E6474C704175B69BCCC762D323F17713EE47F824A2701626DD472811D2726753C4C74855407F10A8E647A0F38DCB6EBFB44700B1F1B10FA374C71A0A10398582D8C78CF32F00B6C6C4C7E837C18F8C3FC9C7ACC39DA06932DEC701030000000100000004000000FC11C3EF5202E5C7E8E215C41119CDC7665E1BE436D7E947D17F338782B2EBC740C4030E91DDCD4784EBD776BE98E147FC11C3EF5202E5C7E8E215C41119CDC70103000000010000000C000000D4BF4232CEBDE9C7D0BE7E4C08EEC3475EA0E1365550E6C7109BA43A9CE1BC47D877E45F386FCA4722B648BF3A00DDC7DC46DAF43669EB4762871F1B4132EEC7203F1C54A5A0D9471C715B0756B6C4C76E08015E4456EB4780867BF00E9093479CEFB7A45BD1DA4700045F335546DA47E8488833D95FCE47FA37CEAF8DC7E047808A3E538A2B91478CCBA9C4549AE84710D21ADC4D2BD3C7CACE6A65C12FE84758C77F25D46EDAC7BCC803061868E647D4BF4232CEBDE9C7D0BE7E4C08EEC347010300000001000000090000005E4BDEE9544BE1C7B4063F2A556BDEC740ED201A063DA2C78BF73A3A6105E2C77E3118868314E34778CC29BF5A88D0C74E56C670F6F1EA4748FD3AA04AF2CBC7D2467E233B58EE47F469ECD904D0EE47F2201D256BCEDCC766EAA2E55DCDEA47F674F5183765D2C7B87012695A8BD447F8BCD625BEEEECC7D44ED5089B3FD1475E4BDEE9544BE1C7B4063F2A556BDEC7010300000001000000070000001467BCDB3A13DEC76EAE0FEE91F4D4C7A2FA82574445D7C7E4B7BBA8DCD0D7C7B8E41A0F71CEE5C72BE1D7D7970AEBC73E1D531F7633EA47E43AEEA49612C9C790471A6B4B4CB547F84F05C0C249C64793DADD6DA726E1C7ECA991DB3650C2C71467BCDB3A13DEC76EAE0FEE91F4D4C701030000000100000006000000FC932D05C421D8474610CF680CA1E5C74085C2762B19AA47E07D7BB9FFBCD147D06ED16B8FFFE347F6403F475314E6476C97A7C840C9C9C7F269C345C341EE47635F137A53E7EBC77EEA3BB27F70E447FC932D05C421D8474610CF680CA1E5C7':::GEOMETRY
									)
								),
								(
									'010600000004000000010300000001000000060000005BDB3E09BF4DEAC7C0CEA23F4E80AE47F35A1A8627D0ECC736B3EB1235E9D4C7545159979A56E14774F027FFF514CAC700A0363E66705E47A6967AE6B12EEC47263996DC2CAAD2C75AD2FA836054ED475BDB3E09BF4DEAC7C0CEA23F4E80AE4701030000000100000007000000E29458C07575DEC7CFD3B65E88C8EEC7FE78A1E897E0E047BD03725E9BD9EEC76EFF7222186DEA47EBF0C71ECB7CE9C7C81D78227523D64776657B1F918EDFC7985769A16099EF47D0F253BD1154BCC768090012A664C5C7D04D8B39BF61D847E29458C07575DEC7CFD3B65E88C8EEC70103000000010000000B000000E2644D9869DCE7C7EC5C97035560DEC71471CDE8B085D2C76C9EF4DBADA5E8C718CE0A9687D1D1C73374572D9B50EFC72084F73B8194E84726147E09D7A7D5C75450D57892C0DD4764B1E33666DECEC7E04FD07C25FAE94784265A71B37CC7C78A04802A80E8E8472AA4921C8F9DEE472809A7F61E7ECF4798B93C399015C747B048C44A9C8CBAC790D598D6BBD7B047E720F138D46CEAC70046F0F1AD36D147E2644D9869DCE7C7EC5C97035560DEC701030000000100000006000000B61A147DD132E2C7805DD1FDEF0A97474282BC49AC4AD9C7604F49186A1497C7C0659DBC43F797C7885438BCE996B2C77690F0EF9DD3E94736D425274A2DEB470C30851EF35BDA47F2637D1C0646E247B61A147DD132E2C7805DD1FDEF0A9747':::GEOMETRY
								)
						)
							AS tab_21275 (col_36483)
				)
		SELECT
			'010200000002000000EA059BEC11A2E0C700ED687E93C28CC72C208A003852EFC7185229F30174CDC7':::GEOMETRY AS col_36484,
			'-Inf':::FLOAT8 AS col_36485;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=empty/setting=no-mutations](https://teamcity.cockroachdb.com/viewLog.html?buildId=2283744&tab=artifacts#/sqlsmith/setup=empty/setting=no-mutations)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dempty%2Fsetting%3Dno-mutations.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/geo/geomfn/affine_transforms.go
+++ b/pkg/geo/geomfn/affine_transforms.go
@@ -118,6 +118,7 @@ func translate(t geom.T, deltas []float64) (geom.T, error) {
 }
 
 // Scale returns a modified Geometry whose coordinates are multiplied by the factors.
+// REQUIRES: len(factors) >= 2.
 func Scale(g geo.Geometry, factors []float64) (geo.Geometry, error) {
 	var zFactor float64
 	if len(factors) > 2 {
@@ -134,8 +135,8 @@ func Scale(g geo.Geometry, factors []float64) (geo.Geometry, error) {
 	)
 }
 
-// ScaleRelativeToOrigin returns a modified Geometry whose coordinates are multiplied by
-// the factors relative to the origin
+// ScaleRelativeToOrigin returns a modified Geometry whose coordinates are
+// multiplied by the factors relative to the origin.
 func ScaleRelativeToOrigin(
 	g geo.Geometry, factor geo.Geometry, origin geo.Geometry,
 ) (geo.Geometry, error) {
@@ -176,6 +177,13 @@ func ScaleRelativeToOrigin(
 		return geo.Geometry{}, errors.Wrap(err, "number of dimensions for the scaling factor and origin must be equal")
 	}
 
+	// This is inconsistent with PostGIS, which allows a POINT EMPTY, but whose
+	// behavior seems to depend on previous queries in the session, and not
+	// desirable to reproduce.
+	if len(originPointG.FlatCoords()) < 2 {
+		return geo.Geometry{}, errors.Newf("the origin must have at least 2 coordinates")
+	}
+
 	// Offset by the origin, scale, and translate it back to the origin.
 	offsetDeltas := make([]float64, 0, 3)
 	offsetDeltas = append(offsetDeltas, -originPointG.X(), -originPointG.Y())
@@ -187,11 +195,18 @@ func ScaleRelativeToOrigin(
 		return geo.Geometry{}, err
 	}
 
-	xFactor, yFactor := factorPointG.X(), factorPointG.Y()
-	var zFactor float64 = 1
-	if factorPointG.Layout().ZIndex() != -1 {
-		zFactor = factorPointG.Z()
+	var xFactor, yFactor, zFactor float64
+	zFactor = 1
+	if len(factorPointG.FlatCoords()) < 2 {
+		// POINT EMPTY results in factors of 0, similar to PostGIS.
+		xFactor, yFactor = 0, 0
+	} else {
+		xFactor, yFactor = factorPointG.X(), factorPointG.Y()
+		if factorPointG.Layout().ZIndex() != -1 {
+			zFactor = factorPointG.Z()
+		}
 	}
+
 	retT, err = affine(
 		retT,
 		AffineMatrix([][]float64{

--- a/pkg/geo/geomfn/affine_transforms_test.go
+++ b/pkg/geo/geomfn/affine_transforms_test.go
@@ -225,6 +225,13 @@ func TestScaleRelativeToOrigin(t *testing.T) {
 			origin:   geom.NewPointFlat(geom.XY, []float64{1, 1}),
 			expected: geom.NewPointEmpty(geom.XY),
 		},
+		{
+			desc:     "scale using an empty point as factor",
+			input:    geom.NewLineStringFlat(geom.XY, []float64{1, 1, 2, 2}),
+			factor:   geom.NewPointEmpty(geom.XY),
+			origin:   geom.NewPointFlat(geom.XY, []float64{1, 1}),
+			expected: geom.NewLineStringFlat(geom.XY, []float64{1, 1, 1, 1}),
+		},
 	}
 
 	for _, tc := range testCases {
@@ -324,6 +331,13 @@ func TestCollectionScaleRelativeToOrigin(t *testing.T) {
 			factor:      geom.NewPointFlat(geom.XY, []float64{2, 2}),
 			origin:      geom.NewLineStringFlat(geom.XY, []float64{1, 1, 1, 1}),
 			expectedErr: "the false origin must be a Point",
+		},
+		{
+			desc:        "scale using an empty point as origin",
+			input:       geom.NewLineStringFlat(geom.XY, []float64{1, 1, 2, 2}),
+			factor:      geom.NewPointFlat(geom.XY, []float64{2, 2}),
+			origin:      geom.NewPointEmpty(geom.XY),
+			expectedErr: "the origin must have at least 2 coordinates",
 		},
 	}
 

--- a/pkg/sql/logictest/testdata/logic_test/geospatial
+++ b/pkg/sql/logictest/testdata/logic_test/geospatial
@@ -4796,27 +4796,28 @@ POLYGON ((-0.1 0, 1 0, 1 1, -0.1 1, -0.1 0))           POLYGON ((0.9 1, 2 1, 2 2
 POLYGON ((-1 0, 0 0, 0 1, -1 1, -1 0))                 POLYGON ((0 1, 1 1, 1 2, 0 2, 0 1))
 POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))                    POLYGON ((1 1, 2 1, 2 2, 1 2, 1 1))
 
-query TTTT
+query TTTTT
 SELECT
   ST_AsText(a.geom) d,
   ST_AsText(ST_Scale(a.geom, .5 , 2)),
   ST_AsText(ST_Scale(a.geom, 'Point(.25 3)')),
-  ST_AsText(ST_Scale(a.geom, 'Point(2 2)', 'Point(1 1)'))
+  ST_AsText(ST_Scale(a.geom, 'Point(2 2)', 'Point(1 1)')),
+  ST_AsText(ST_Scale(a.geom, 'POINT EMPTY'))
 FROM geom_operators_test a
 ORDER BY d ASC
 ----
-NULL                                                   NULL                                                   NULL                                                      NULL
-GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))  GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))  GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))     GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (-1 -1)))
-GEOMETRYCOLLECTION EMPTY                               GEOMETRYCOLLECTION EMPTY                               GEOMETRYCOLLECTION EMPTY                                  GEOMETRYCOLLECTION EMPTY
-LINESTRING (-0.5 0.5, 0.5 0.5)                         LINESTRING (-0.25 1, 0.25 1)                           LINESTRING (-0.125 1.5, 0.125 1.5)                        LINESTRING (-2 0, 0 0)
-LINESTRING EMPTY                                       LINESTRING EMPTY                                       LINESTRING EMPTY                                          LINESTRING EMPTY
-POINT (-0.5 0.5)                                       POINT (-0.25 1)                                        POINT (-0.125 1.5)                                        POINT (-2 0)
-POINT (0.5 0.5)                                        POINT (0.25 1)                                         POINT (0.125 1.5)                                         POINT (0 0)
-POINT (5 5)                                            POINT (2.5 10)                                         POINT (1.25 15)                                           POINT (9 9)
-POINT EMPTY                                            POINT EMPTY                                            POINT EMPTY                                               POINT EMPTY
-POLYGON ((-0.1 0, 1 0, 1 1, -0.1 1, -0.1 0))           POLYGON ((-0.05 0, 0.5 0, 0.5 2, -0.05 2, -0.05 0))    POLYGON ((-0.025 0, 0.25 0, 0.25 3, -0.025 3, -0.025 0))  POLYGON ((-1.2 -1, 1 -1, 1 1, -1.2 1, -1.2 -1))
-POLYGON ((-1 0, 0 0, 0 1, -1 1, -1 0))                 POLYGON ((-0.5 0, 0 0, 0 2, -0.5 2, -0.5 0))           POLYGON ((-0.25 0, 0 0, 0 3, -0.25 3, -0.25 0))           POLYGON ((-3 -1, -1 -1, -1 1, -3 1, -3 -1))
-POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))                    POLYGON ((0 0, 0.5 0, 0.5 2, 0 2, 0 0))                POLYGON ((0 0, 0.25 0, 0.25 3, 0 3, 0 0))                 POLYGON ((-1 -1, 1 -1, 1 1, -1 1, -1 -1))
+NULL                                                   NULL                                                   NULL                                                      NULL                                                     NULL
+GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))  GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))  GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))     GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (-1 -1)))  GEOMETRYCOLLECTION (GEOMETRYCOLLECTION (POINT (0 0)))
+GEOMETRYCOLLECTION EMPTY                               GEOMETRYCOLLECTION EMPTY                               GEOMETRYCOLLECTION EMPTY                                  GEOMETRYCOLLECTION EMPTY                                 GEOMETRYCOLLECTION EMPTY
+LINESTRING (-0.5 0.5, 0.5 0.5)                         LINESTRING (-0.25 1, 0.25 1)                           LINESTRING (-0.125 1.5, 0.125 1.5)                        LINESTRING (-2 0, 0 0)                                   LINESTRING (0 0, 0 0)
+LINESTRING EMPTY                                       LINESTRING EMPTY                                       LINESTRING EMPTY                                          LINESTRING EMPTY                                         LINESTRING EMPTY
+POINT (-0.5 0.5)                                       POINT (-0.25 1)                                        POINT (-0.125 1.5)                                        POINT (-2 0)                                             POINT (0 0)
+POINT (0.5 0.5)                                        POINT (0.25 1)                                         POINT (0.125 1.5)                                         POINT (0 0)                                              POINT (0 0)
+POINT (5 5)                                            POINT (2.5 10)                                         POINT (1.25 15)                                           POINT (9 9)                                              POINT (0 0)
+POINT EMPTY                                            POINT EMPTY                                            POINT EMPTY                                               POINT EMPTY                                              POINT EMPTY
+POLYGON ((-0.1 0, 1 0, 1 1, -0.1 1, -0.1 0))           POLYGON ((-0.05 0, 0.5 0, 0.5 2, -0.05 2, -0.05 0))    POLYGON ((-0.025 0, 0.25 0, 0.25 3, -0.025 3, -0.025 0))  POLYGON ((-1.2 -1, 1 -1, 1 1, -1.2 1, -1.2 -1))          POLYGON ((0 0, 0 0, 0 0, 0 0, 0 0))
+POLYGON ((-1 0, 0 0, 0 1, -1 1, -1 0))                 POLYGON ((-0.5 0, 0 0, 0 2, -0.5 2, -0.5 0))           POLYGON ((-0.25 0, 0 0, 0 3, -0.25 3, -0.25 0))           POLYGON ((-3 -1, -1 -1, -1 1, -3 1, -3 -1))              POLYGON ((0 0, 0 0, 0 0, 0 0, 0 0))
+POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))                    POLYGON ((0 0, 0.5 0, 0.5 2, 0 2, 0 0))                POLYGON ((0 0, 0.25 0, 0.25 3, 0 3, 0 0))                 POLYGON ((-1 -1, 1 -1, 1 1, -1 1, -1 -1))                POLYGON ((0 0, 0 0, 0 0, 0 0, 0 0))
 
 query TT
 SELECT

--- a/pkg/sql/sem/builtins/geo_builtins.go
+++ b/pkg/sql/sem/builtins/geo_builtins.go
@@ -4035,7 +4035,13 @@ The matrix transformation will be applied as follows for each coordinate:
 					return nil, errors.Newf("a Point must be used as the scaling factor")
 				}
 
-				ret, err := geomfn.Scale(g.Geometry, pointFactor.FlatCoords())
+				factors := pointFactor.FlatCoords()
+				if len(factors) < 2 {
+					// Scale by 0, and leave Z untouched (this matches the behavior of
+					// ScaleRelativeToOrigin).
+					factors = []float64{0, 0, 1}
+				}
+				ret, err := geomfn.Scale(g.Geometry, factors)
 				if err != nil {
 					return nil, err
 				}

